src/platform_impl/windows/event_loop.rs

```diff
@@ -667,15 +751,34 @@ unsafe extern "system" fn public_window_callback<T: 'static>(
                 window_id: RootWindowId(WindowId(window)),
                 event: Destroyed,
             });
+            subclass_input.event_loop_runner.remove_window(window);
 
             drop(subclass_input);
             Box::from_raw(subclass_input_ptr as *mut SubclassInput<T>);
             0
         }
 
         winuser::WM_PAINT => {
-            subclass_input.event_loop_runner.main_events_cleared();
-            subclass_input.send_event(Event::RedrawRequested(RootWindowId(WindowId(window))));
+            if subclass_input.event_loop_runner.should_buffer() {
+                panic!("InvalidateRgn should be used instead of UpdateWindow");
```

Comment:
we should never panic in a Windows callback as it will cause Rust to unwind into foreign code which is undefined behavior

Reply:
Good catch! I've wrapped the entire match block in a `catch_unwind` call, so we shouldn't have to worry about this any more.